src/reducer.js

```diff
@@ -0,0 +1,34 @@
+import { keyBy, without, uniq, omit } from 'lodash'
+import { SET, ADD, DEL, RESET } from './actions'
+
+export const initState = { datas: {}, keys: [], nb: 0, initialized: false }
+
+export default key => prefix =>
+  (state = initState, { type = 'UNKONWN', payload } = {}) => {
+    switch (type) {
+      case SET(prefix):
+        return {
+          datas: keyBy(payload, key),
+          keys: payload.map(element => element[key]),
+          nb: payload.length,
+          initialized: true,
+        }
+      case ADD(prefix):
+        return {
+          ...state,
```

Comment:
it's not possible to add data the first time ? initialized properties is set only on SET action ?